### DISCUSSION TITLE
How to return all error at once ?

### DISCUSSION THREAD

**Discussion by iamshulz** [fastify/fastify#3386] [Q&A] [answered]
I have created an errorHandler but I noticed that it only returns one error message even if there are multiple valiolations.

```js
  // Error Handler
  fastify.setErrorHandler(function (error, request, reply) {
    if (error.validation) {
      reply.status(400).send(error.validation);
    }

    reply.send(error)
  });


//Post
const createUser = {
  body: {
    type: "object",
    properties: { message: { type: "string" }, item: { type: "string" } },
    required: [ "item","message",],
  },
};

// Sample input

{
    "age": "Hello World"   
}
```

//Error result
```js
[
    {
        "keyword": "required",
        "dataPath": "",
        "schemaPath": "#/required",
        "params": {
            "missingProperty": "message"
        },
        "message": "should have required property 'message'"
    }
]
```

**Comment by delvedor** [comment, marked as answer]
You can configure Ajv to do that, it was the default value some time ago, but then we discovered that having `allErrors` set to `true` will make you at risk of DoS attacks, see the [CVE](https://snyk.io/vuln/SNYK-JS-FASTIFY-595959).

tl;dr: we don't recommend it!

**Reply by iamshulz** [reply]
that meanss I have to install ajv and this doesn't work anymore? 
```
 const app = Fastify({
    logger: true,
    ajv: {
      customOptions: { jsonPointers: true },
      plugins: [
        require('ajv-errors')
      ]
    }
  });
  ```

**Reply by delvedor** [reply]
You can configure ajv: https://www.fastify.io/docs/latest/Server/#ajv
